tests/capi/unittest_capi_service.cc

```diff
@@ -49,6 +203,164 @@ TEST (nnstreamer_capi_service, basic_test_0_p)
+      "appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_filter name=filter_h framework=tensorflow-lite model=%s ! tensor_sink name=tensor_sink",
+      test_model);
+
+  ret = ml_service_set_pipeline(key, pipeline);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  ret = ml_service_get_pipeline(key, &ret_pipeline);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  /* Test */
+  ret = ml_service_del_pipeline (key);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  ret = ml_service_get_pipeline(key, &ret_pipeline);
+  EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, ret);
+
+  g_free (pipeline);
+  g_free (ret_pipeline);
+  g_free (test_model);
+}
+
+/**
+ * @brief Delete the pipeline description with wrong parameters
+ */
+TEST (nnstreamer_capi_service, del_pipeline_description_1_n)
+{
+  int ret = 0;
+  gchar *pipeline, *test_model;
+  gchar *ret_pipeline = NULL;
+  const gchar *key = "ServiceName";
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (
+      root_path, "tests", "test_models", "models", "add.tflite", NULL);
+  EXPECT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  pipeline = g_strdup_printf (
+      "appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_filter name=filter_h framework=tensorflow-lite model=%s ! tensor_sink name=tensor_sink",
+      test_model);
+
+  ret = ml_service_set_pipeline (key, pipeline);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  /* Test */
+  ret = ml_service_del_pipeline (NULL);
+  EXPECT_EQ (ML_ERROR_INVALID_PARAMETER, ret);
+
+  g_free (pipeline);
+  g_free (ret_pipeline);
+  g_free (test_model);
+}
+
+/**
+ * @brief Delete the pipeline description twice and check its result
+ */
+TEST (nnstreamer_capi_service, del_pipeline_description_2_n)
+{
+  int ret = 0;
+  gchar *pipeline, *test_model;
+  const gchar *key = "ServiceName";
+  const gchar *root_path = g_getenv ("NNSTREAMER_SOURCE_ROOT_PATH");
+
+  /* supposed to run test in build directory */
+  if (root_path == NULL)
+    root_path = "..";
+
+  test_model = g_build_filename (
+      root_path, "tests", "test_models", "models", "add.tflite", NULL);
+  EXPECT_TRUE (g_file_test (test_model, G_FILE_TEST_EXISTS));
+
+  pipeline = g_strdup_printf (
+      "appsrc name=appsrc ! "
+      "other/tensor,dimension=(string)1:1:1:1,type=(string)float32,framerate=(fraction)0/1 ! "
+      "tensor_filter name=filter_h framework=tensorflow-lite model=%s ! tensor_sink name=tensor_sink",
+      test_model);
+
+  ret = ml_service_set_pipeline(key, pipeline);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  ret = ml_service_del_pipeline (key);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
+
+  /* Test */
+  ret = ml_service_del_pipeline (key);
+  EXPECT_EQ (ML_ERROR_NONE, ret);
```

Comment:
Is it negative case?

Reply:
You are right. It is not a negative test case. I will merge this into the positive one.